backend/src/components/student.js

```diff
@@ -102,29 +102,31 @@ async function getStudentByPen(req, res) {
  */
 async function createNewStudent(req, res) {
   try {
-    let transactionId;
+    let transactionID;
     if (req.session.create_new_student_transactionID) {
-      transactionId = req.session.create_new_student_transactionID;
+      transactionID = req.session.create_new_student_transactionID;
     } else {
-      transactionId = uuidv4();
-      req.session.create_new_student_transactionID = transactionId; // store it in session so that it can be reused when the api call to create student fails.
+      transactionID = uuidv4();
+      req.session.create_new_student_transactionID = transactionID; // store it in session so that it can be reused when the api call to create student fails.
     }
     const params = {
       params: {
-        transactionId
+        transactionID
       }
     };
     const token = utils.getBackendToken(req);
     const penNumber = await getData(token, config.get('server:penServices:nextPenURL'), params);
     const student = req.body.student;
     student.pen = penNumber;
+    student.sexCode = student.genderCode; // sex code is mandatory in API.
```

Comment:
There is a description about Sex Code in my ticket, `IF request Gender Code = X, set Sex Code to U; ELSE set Sex Code to the same as request Gender Code.`. Not sure if it is also required here.

Reply:
Ok  I will update the code to set it in UI, so that it is page specific